AWSAuthSDK/AWSAuthSDKTestAppUITests/AWSHostedUIUserPoolTests.swift

```diff
@@ -14,7 +14,88 @@ class AWSHostedUIUserPoolTests: AWSAuthSDKUITestBase {
         continueAfterFailure = false
         XCUIApplication().launch()
     }
-    
+
+    /// Test successful authentication in user pool using hosted UI
+    ///
+    /// - Given: A verified user
+    /// - When:
+    ///    - I try to sign in using hosted UI
+    /// - Then:
+    ///    - I should get a signed in session
+    ///
+    func testHostedUISignIn() {
+        let username = "testUser" + UUID().uuidString
+        let app = XCUIApplication()
+        signOutUserpool(application: app)
+        
+        signUpAndVerifyUser(username: username)
+        
+        // Push the hosted UI view controller
+        app.buttons["Hosted UI Userpool tests"].tap()
+        XCTAssertTrue(app.navigationBars["UserPool"].exists)
+        
+        //Initiate signIn
+        app.buttons["SignIn User"].tap()
+        signInUserpool(application: app, username: username)
+        
+        // Check if successfully signed in
+        let userPoolSignInStateLabelElement = app.staticTexts["userPoolSignInStateLabel"]
+        let predicate = NSPredicate(format: "label CONTAINS[c] %@", "signedIn")
+        let signInExpectation = expectation(for: predicate,
+                                            evaluatedWith: userPoolSignInStateLabelElement,
+                                            handler: nil)
+        wait(for: [signInExpectation], timeout: 5)
+    }
+
+    /// Test successful re-authentication in user pool using hosted UI
+    ///
+    /// - Given: A signed In user
+    /// - When:
+    ///    - I try invalidate the refresh token and sign In again
+    /// - Then:
+    ///    - I should get a signed in session
+    ///
+    func testHostedUISignInAfterRefreshTokenInvalid() {
+        let username = "testUser" + UUID().uuidString
+        let app = XCUIApplication()
+        signOutUserpool(application: app)
+        
+        signUpAndVerifyUser(username: username)
+        
+        // Push the hosted UI view controller
+        app.buttons["Hosted UI Userpool tests"].tap()
+        XCTAssertTrue(app.navigationBars["UserPool"].exists)
+        
+        //Initiate signIn
+        app.buttons["SignIn User"].tap()
+        signInUserpool(application: app, username: username)
+        
+        // Check if successfully signed in
+        let userPoolSignInStateLabelElement = app.staticTexts["userPoolSignInStateLabel"]
+        let predicate = NSPredicate(format: "label CONTAINS[c] %@", "signedIn")
+        let signInExpectation = expectation(for: predicate,
+                                            evaluatedWith: userPoolSignInStateLabelElement,
+                                            handler: nil)
+        wait(for: [signInExpectation], timeout: 5)
```

Comment:
[low] since it's duplicated, can you just call `testHostedUISignIn()`? or probably need to introduce a new private method that returns whatever you need to perform the rest of the test, or pass in what you use later, ie. the username. but if there aren't other scenarios to test and the new private method is just reused once, then can also just keep it the way it is

Reply:
Moved the duplicate code